sharktank_models/llama3.1/test_llama.py

```diff
@@ -106,34 +144,48 @@ def cross_entropy(predictions, targets, epsilon=1e-12):
 
 
 def prefill_cross_entropy(toy_llama, ids):
-    if not hasattr(toy_llama, "prefill"):
-        return
+    logits = toy_llama.prefill(ids=ids)
+    ids = numpy.asarray([ids])
 
-    assert ids.shape[1] % block_size == 0
+    logits = logits[0, :-1, :]
+    ids = ids[:, 1:]
 
-    len = numpy.asarray([ids.shape[1]], numpy.int64)
-    pages = ids.shape[1] // block_size
-    pages = numpy.asarray([list(range(pages))], numpy.int64)
-    cache = numpy.zeros((128, page_size), numpy.float16)
+    return cross_entropy(logits, ids)
 
-    logits = toy_llama.prefill(ids[:, : ids.shape[1]], len, pages, cache)
 
-    logits = numpy.asarray(logits).astype(numpy.float32)
-    ids = numpy.asarray(ids)
+def decode_cross_entropy(toy_llama, ids):
+    N = len(ids)
+    logits = toy_llama.decode(ids=ids, start=0)
+    ids = numpy.asarray([ids])
+    logits = logits[0, :-1, :]
+    ids = ids[:, 1:]
+
+    return cross_entropy(logits, ids)
```

Comment:
Out of curiosity, what does `logits` look like before this slicing? What information is discarded?

Reply:
Added a comment to clarify. Its just comparing the predicted to the selected token sequence.